flaml/autogen/agentchat/responsive_agent.py

```diff
@@ -262,19 +264,27 @@ def _append_oai_message(self, message: Union[Dict, str], role, conversation_id:
 
         Args:
             message (dict or str): message to be appended to the ChatCompletion conversation.
-            role (str): role of the message, can be "assistant" or "function".
+            role (str): role of the message, can be "assistant", "user" or "function", this will overwrite the `role` in the message only when original role is not "function".
             conversation_id (Agent): id of the conversation, should be the recipient or sender.
 
         Returns:
             bool: whether the message is appended to the ChatCompletion conversation.
         """
         message = self._message_to_dict(message)
         # create oai message to be appended to the oai conversation that can be passed to oai directly.
-        oai_message = {k: message[k] for k in ("content", "function_call", "name", "context") if k in message}
-        if "content" not in oai_message and "function_call" not in oai_message:
+        oai_message = {k: message[k] for k in ("content", "function_call", "name", "role", "context") if k in message}
+        # 'content' is a required field
+        if "content" not in oai_message:
+            return False
+        if role is not None:
+            if oai_message["role"] is "function" and role is not "function":
+                print(f"Warning: Attempt to overwrite role 'function' with {role}. Rejected.")
+                return False
```

Comment:
This will make the result to be regarded as invalid and prevent proper processing of function call results.

Reply:
Thank you for pointing out. I have revised it.